pkg/controller/platformadmin/platformadmin_controller.go

```diff
@@ -600,3 +640,115 @@ func annotationToComponent(annotation map[string]string) ([]*config.Component, e
+		return nil, err
+	}
+
+	// If PlatformAdmin is about to be deleted, remove Finalizer from the framework.
+	// If not deleted, the owner reference is synchronized.
+	if platformAdmin.DeletionTimestamp != nil {
+		_, err = controllerutil.CreateOrUpdate(ctx, r.Client, cm, func() error {
+			// During the deletion phase, ensure that data in the framework is read before deletion
+			// The following code removes the finalizer, allowing the framework to be deleted (since we read out its data above).
+			controllerutil.RemoveFinalizer(cm, FrameworkFinalizer)
+			return nil
+		})
+		if err != nil {
+			klog.Errorf(Format("Failed to remove finalizer of framework configmap for PlatformAdmin %s/%s", platformAdmin.Namespace, platformAdmin.Name))
+			return nil, err
+		}
+	} else {
+		hasOwnerReference := false
+		for _, ref := range cm.ObjectMeta.OwnerReferences {
+			if ref.Kind == platformAdmin.Kind && ref.Name == platformAdmin.Name {
+				hasOwnerReference = true
+			}
+		}
+		if !hasOwnerReference {
+			_, err = controllerutil.CreateOrUpdate(ctx, r.Client, cm, func() error {
+				return controllerutil.SetOwnerReference(platformAdmin, cm, r.scheme)
+			})
+			if err != nil {
+				klog.Errorf(Format("Failed to add owner reference of framework configmap for PlatformAdmin %s/%s", platformAdmin.Namespace, platformAdmin.Name))
+				return nil, err
+			}
+		}
+	}
+
+	return platformAdminFramework, nil
+}
+
+// initFramework initializes the framework information for PlatformAdmin
+func (r *ReconcilePlatformAdmin) initFramework(ctx context.Context, platformAdmin *iotv1alpha2.PlatformAdmin, platformAdminFramework *PlatformAdminFramework) error {
+	klog.V(6).Infof(Format("Initializes the standard framework information for PlatformAdmin %s/%s", platformAdmin.Namespace, platformAdmin.Name))
+
+	// Use standard configurations to build the framework
+	platformAdminFramework.security = platformAdmin.Spec.Security
+	if platformAdminFramework.security {
+		platformAdminFramework.ConfigMaps = r.Configration.SecurityConfigMaps[platformAdmin.Spec.Version]
+		platformAdminFramework.Components = r.Configration.SecurityComponents[platformAdmin.Spec.Version]
+	} else {
+		platformAdminFramework.ConfigMaps = r.Configration.NoSectyConfigMaps[platformAdmin.Spec.Version]
+		platformAdminFramework.Components = r.Configration.NoSectyComponents[platformAdmin.Spec.Version]
+	}
+
+	// For better serialization, the serialization method of the Kubernetes runtime library is used
+	data, err := runtime.Encode(r.yamlSerializer, platformAdminFramework)
+	if err != nil {
+		klog.Errorf(Format("Failed to marshal framework for PlatformAdmin %s/%s", platformAdmin.Namespace, platformAdmin.Name))
+		return err
+	}
+
+	// Create the configmap that represents framework
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      platformAdminFramework.name,
+			Namespace: platformAdmin.Namespace,
+		},
+	}
+	cm.Labels = make(map[string]string)
```

Comment:
```
        cm.Labels[iotv1alpha2.LabelPlatformAdminGenerate] = LabelFramework
	cm.Data = make(map[string]string)
	cm.Data["framework"] = string(data)
```
The above part of the code needs to be placed in the CreateOrUpdate function, otherwise configmap cannot be updated.